app/src/components/common/EmailInputWithDomainBasedSuggestions/index.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import _ from "lodash";
+import { getFunctions, httpsCallable } from "firebase/functions";
+import { useSelector } from "react-redux";
+import { getUserAuthDetails } from "store/selectors";
+import { getDomainFromEmail, isCompanyEmail, isEmailValid } from "utils/FormattingHelper";
+import CreatableSelect from "react-select/creatable";
+import { MultiValue } from "react-select";
+
+interface Props {
+  onChange: (emails: string[]) => void;
+  transparentBackground?: boolean;
+}
+
+const EmailInputWithDomainBasedSuggestions: React.FC<Props> = ({ onChange, transparentBackground = false }) => {
+  const user = useSelector(getUserAuthDetails);
+  const userEmail = user?.details?.profile?.email;
+
+  const [allSuggestions, setAllSuggestions] = useState<string[]>([]);
+
+  const getOrganizationUsers = useMemo(() => httpsCallable(getFunctions(), "users-getOrganizationUsers"), []);
+
+  useEffect(() => {
+    if (!isCompanyEmail(userEmail)) return;
```

Comment:
We should also check if the user is a `verified` business domain user

Reply:
We already have a verified check in backend `users-getOrganizationUsers`